src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1058,31 +1321,124 @@ def _set_up_defaults(
+        custom_location: "CustomLocation",
+):
+    if not env.resource_type or (env.is_connected_environment() and (not env.name or not resource_group.name or not env.custom_location_id)):
+        connected_env_list = list_connected_environments(cmd=cmd, resource_group_name=resource_group.name)
+        env_list = []
+        for e in connected_env_list:
+            if env.name and env.name != e["name"]:
+                continue
+            if location and format_location(e["location"]) != location:
+                continue
+            if custom_location.name and e["extendedLocation"]["name"].lower() != custom_location.get_rid().lower():
+                continue
+            if custom_location.connected_cluster_id:
+                custom_location_from_env = get_custom_location(cmd=cmd, custom_location_id=e["extendedLocation"]["name"])
+                if custom_location_from_env is None or custom_location.connected_cluster_id.lower() != custom_location_from_env.host_resource_id.lower():
+                    continue
+            env_list.append(e)
+
+        if len(env_list) == 1:
+            resource_group.name = parse_resource_id(env_list[0]["id"])["resource_group"]
+            env.set_name(env_list[0]["id"])
+            env.custom_location_id = env_list[0]["extendedLocation"]["name"]
+        if len(env_list) > 1:
+            if env.name:
+                raise ValidationError(
+                    f"There are multiple environments with name {env.name} on the subscription. "
+                    "Please specify which resource group your Connected environment is in."
+                )  # get ACR details from --image, if possible
+            else:
+                resource_group.name = parse_resource_id(env_list[0]["id"])["resource_group"]
+                env.set_name(env_list[0]["id"])
+                env.custom_location_id = env_list[0]["extendedLocation"]["name"]
+
+
+def _infer_existing_custom_location_or_extension(
+        cmd,
+        name,
+        location,
+        resource_group: "ResourceGroup",
+        env: "ContainerAppEnvironment",
+        custom_location: "CustomLocation",
+        extension: "Extension"
+):
+    # Try to get existed custom location and extension
+    # Set up default values for not existed resources(env, custom location, extension)
+    if env.is_connected_environment():
+        if not env.check_exists():
+            _prepare_custom_location_args(cmd, custom_location=custom_location, extension=extension)
+            if location is None:
+                env.location = custom_location.location
+                resource_group.location = custom_location.location
+            if custom_location.exists:
+                env.custom_location_id = custom_location.get_rid()
+            else:
+                random_int = randint(0, 9999)
+                resource_group.name = get_randomized_name(get_profile_username(), name=resource_group.name, random_int=random_int)
```

Comment:
not respect customer input rg?

Reply:
It will respect the customer input rg if `resource_group.name` exist.
```
def get_randomized_name(prefix, name=None, initial="rg", random_int=None):
    from random import randint
    random_int = random_int if random_int else randint(0, 9999)
    default = "{}_{}_{:04}".format(prefix, initial, random_int)
    if name is not None:
        return name
    return default
```

Reply:
then please move out of generate randomized name function. get_randomized_name() should only generate random name

Reply:
Done.